### ISSUE TITLE
 Build 2.5 on Windows fails using VC2017 community

### ISSUE THREAD

**Issue by zhao07** [fireice-uk/xmr-stak#1905]
compile cpu, opencl OK, but when compiling CUDA，it says

“C:\projects\xmr-stak\build-yet\install.vcxproj”(1) ->
“C:\projects\xmr-stak\build\ALL_BUILD.vcxproj” (3) ->
“C:\projects\xmr-stak\build\xmrstak_cuda_backend.vcxproj”(8) ->
(CustomBuild) ->
  c:\program files\nvidia gpu computing toolkit\cuda\v9.1\include\crt/host_config.h(135): fatal error C1189: #error:  -
- unsupported Microsoft Visual Studio version! Only the versions 2012, 2013, 2015 and 2017 are supported! [C:\projects\
xmr-stak\build\xmrstak_cuda_backend.vcxproj]

How to do ？
using VC2017 community， CUDA 9.1

**Comment by psychocrypt** [comment]
This is a message from CUDA. CUDA 9.1 is not compatible with your MSVC.
Please update cuda to version 9.2 or 10.

**Comment by SChernykh** [comment]
@psychocrypt I had the same problem with CUDA 9.2 and VS2017. Visual Studio updates internal versio number like every 2 weeks, and CUDA 9.2 doesn't recognize now as a result. I just fixed version number for VS2017 in host_config.h and it all compiled fine.

**Comment by psychocrypt** [comment]
Yes NVIDIA is hard coding supported VS versions. If the minor number of VS
changed it is not supported anymore from old CUDA versions.
